docs/testplan/latency_test.md

```diff
@@ -0,0 +1,99 @@
+
+   - Similarly, in IXIA-2, define traffic items targeting IXIA-1 and IXIA-3.
+   - Repeat the pattern across all traffic generators to establish a robust traffic matrix.
+
+## Test Case 1: Full-mesh Latency Test
+
+1. **Baseline Latency Measurement**
+   - Start all the traffic items simultaneously and run them for 1 minute. Record latency statistics.
+   - Record latency statistics from each traffic item.
+   - Compare latency differences between the two traffic items sourced from each traffic generator.
+2. **Packet Size Variation**
+   - Repeat the baseline test using different frame sizes: 64 bytes, 1024 bytes, 4096 bytes, and 8192 bytes.
+   - Analyze how packet size affects latency.
+3. **Traffic Rate Scaling**
+   - Increase each traffic item’s rate from 60% to 70%, 80%, and 90% of the line rate.
+   - For each traffic rate, repeat the baseline and packet size variation tests.
+   - Observe how latency and packet loss change with increased traffic load.
+4. **RFC2889 Mode Evaluation**
+   - Enable the RFC2889 flag in the traffic generator settings.
+   - Repeat all previous tests under this mode.
+   - Measure whether this setting has any impact on latency or packet loss.
+
+## Test case 2: Latency Test with an Oversubscribed Port
+
+This test is conducted in a one-tier network to evaluate latency under an oversubscribed state.
+
+1. Assume the DUT has X ports connected to traffic generators. Assume the DUT has X ports connected to traffic generators. Select the last port on a traffic generator as the Rx port, and designate the remaining (X-1) ports as Tx ports.
+2. Define (X-1) traffic items, each assigned to a different Tx port, using a frame size of 86 bytes. Set each traffic stream's rate to `(line_rate × 110%) / X` to create an oversubscribed condition.
+3. Start all the traffic items simultaneously and run them for 1 minute. Record latency statistics.
+4. Repeat the test with frame size 1024 bytes, 4096 bytes, and 8192 bytes.
+5. Analyze the results by comparing latency measurements across different frame sizes, identifying any patterns or anomalies, and determining how oversubscription affects latency performance.
+
+## Metrics Processing
+
+Latency data is collected and stored periodically. The diagram below illustrates how it is organized in the database. For more details, refer to `test_reporting/telemetry/README.md`.
+
+![metrics](./datapoints.png)
+
+For each of the above results, report the latency figures in nanoseconds to a database via the telemetry FinalMetricsReporter provided by the SONiC team. An example of how to use the interface is provided in telemetry folder. The metrics are stored as data points in our database.
+
+| User Interface Label                     | Label Key in DB          | Example Value       |
+| ---------------------------------------- | ------------------------ | ------------------- |
+| `METRIC_LABEL_DEVICE_ID`                 | device.id                | switch-A            |
```

Comment:
if we are limitting the test to 1 single device, we should make it clear in the topology setup.

the current topology setup is suggesting we support multiple ones.